### PR TITLE
Fix safelogin for 1.18 and above by using worldProvider minHeight

### LINKED ISSUE #4665: Under y=0 Fly gets reenabled automatically
### Type of bug

Other unexpected behaviour

### `/ess dump all` output

https://essentialsx.net/dump.html?id=c041cea1f72941fbbffdc7bad17c00d4

### Error log (if applicable)

_No response_

### Bug description

While opped, fly mode gets reenabled every time I rejoin the server under a certain y level. This does not happen normally, only under y=-1. This command "essentials.fly.safelogin" may be doing it, it describes itself as only enabling on rejoin if you're in the air, so I would assume that it thinks everything under y=-1 is void since 1.18 added new stuff.

### Steps to reproduce

1. Start as a server op.
2. Make a hole down to y=-1
3. Go into survival mode.
4. Make sure fly is already toggled off and try rejoining.

### Expected behaviour

It should log me back in without enabling fly mode unless I'm currently in the air.

### Actual behaviour

It thinks I'm in the void at y=-1 and below so it is always enabling fly regardless.

### PR DESCRIPTION
**Pull request by Pirgosth**
### Information

This PR fixes #4665 . 

### Details

**Proposed fix:**    

There was already a partial fix for this (#4667) but the same improvement was missing for the else statement of the shouldFly function in LocationUtil.

We can now login below y=0 (tested with y=-63) and it does not enable fly mode if the player is actually in a safe location. It also fixed the safelogin when falling into the void for y <= -64 for versions >= 1.18.
It does not break backward compatibility for older versions of Minecraft.

**Environments tested:**    

OS: Fedora 35

Java version:  "17.0.1" 2021-10-19 LTS

- [ ] Most recent Paper version (1.XX.Y, git-Paper-BUILD)
- [ ] CraftBukkit/Spigot/Paper 1.12.2
- [ ] CraftBukkit 1.8.8
- [x] Spigot 1.18.1 (CraftBukkit version 3368-Spigot-8965a50-2a2caa7 (MC: 1.18.1) (Implementing API version 1.18.1-R0.1-SNAPSHOT))

**Demonstration:**    

As you can see we are no longer in fly mode just after logged in. (I ensure myself that fly mode was disabled before logging in back)

### CODE REVIEW

File: Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java [hunk 1]
```diff
@@ -266,6 +266,7 @@ public static boolean shouldFly(IEssentials ess, final Location loc) {
         int y = (int) Math.round(loc.getY());
         final int z = loc.getBlockZ();
         int count = 0;
+        //Check whether more than 2 unsafe block are below player.
```
Comment (pop4959) [inline Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java]: Please follow the existing conventions for code comments (space between // and comment).

File: Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java [hunk 2]
```diff
@@ -274,7 +275,8 @@ public static boolean shouldFly(IEssentials ess, final Location loc) {
             }
         }
 
-        return y < 0;
+        //If not then check if player is not in void
```
Comment (pop4959) [inline Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java]: Don't you mean the opposite here? It looks like this is checking if the player *is* in the void.

**Review by pop4959 — CHANGES_REQUESTED** [review]
Change looks good but some minor comments, otherwise LGTM.

File: Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java (hunk 2 above)
Comment (Pirgosth) [reply Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java]: Yes my bad, I should have think twice about it before pushing it, I'll change that

File: Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java (hunk 1 above)
Comment (Pirgosth) [reply Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java]: Yes of course, I'll read them next time I fix a bug, I wrote that quickly and I should have probably take more time for all theses details

File: Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java [hunk 3]
```diff
@@ -274,7 +275,8 @@ public static boolean shouldFly(IEssentials ess, final Location loc) {
             }
         }
 
-        return y < 0;
+        //If not then check if player is in the void
```
Comment (pop4959) [inline Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java]: Sorry, can you also fix this comment :)

**Review by pop4959 — CHANGES_REQUESTED** [review]

File: Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java (hunk 3 above)
Comment (Pirgosth) [reply Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java]: Ahah yes it is already fixed, I've read your comment like a robot, I realized too late that I did forget this one :D

Comment (pop4959) [reply Essentials/src/main/java/com/earth2me/essentials/utils/LocationUtil.java]: No problem, thanks

**Review by pop4959 — APPROVED** [review]

**Comment by pop4959** [comment]
Thanks, appreciate the bugfix :)

**Comment by Pirgosth** [comment]
> Thanks, appreciate the bugfix :)

It was a pleasure, hope I'll have other opportunities to contribute to these projects :)

**Pull request merged**